### PR TITLE
Switch to Monix Observables from Akka Streams

### PR DESCRIPTION
**Pull request by afoerster**
The original intention was to fix an error where the Akka stream would
not terminate when a stream failed (like when it overflowed). I was able
eventually to get the stream to fail and notify the application so the
application could exit. I was unable to get this to work with Akka,
hence the library change.

The Monix code looks essentially the same, but has fewer 'knobs', is
simpler,  and is
better documented. It is also a smaller library that has active support.

The stream failure and exit is handled by a callback. When the
default 'fail' mode is chosen for an overloaded stream, a method is
called that includes a call to `System.exit`. This behavior is tested in
a unit test.

There are also a few style changes included in the changeset.

### CODE REVIEW

File: log-collector/src/main/scala/io/phdata/pulse/logcollector/SolrCloudStream.scala [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      .subscribe()
+
+  /**
+   * Put an event onto the stream
+   *
+   * @param applicationName Name of the application
+   * @param data            The data object, key value pairs
+   */
+  private[logcollector] def put(applicationName: String, data: Map[String, String]): Unit =
+    subject.onNext((applicationName, data)).onComplete(x => x.isSuccess)
```
Comment (bpmcd) [inline log-collector/src/main/scala/io/phdata/pulse/logcollector/SolrCloudStream.scala]: What's the purpose of the `.onComplete` here?  It doesn't appear to return a value or log info.

File: log-collector/src/main/scala/io/phdata/pulse/logcollector/PulseKafkaConsumer.scala [hunk 2]
```diff
@@ -55,7 +55,7 @@ class PulseKafkaConsumer(solrService: SolrService) extends JsonSupport with Lazy
           logger.trace("KAFKA: Consuming " + record.value() + " from topic: " + topic)
           val logEvent    = record.value().parseJson.convertTo[LogEvent]
           val logEventMap = Util.logEventToFlattenedMap(logEvent)
-          solrInputStream ! (logEventMap.getOrElse("application", None), logEventMap)
+          solrInputStream.put(logEventMap.getOrElse("application", ""), logEventMap)
```
Comment (bpmcd) [inline log-collector/src/main/scala/io/phdata/pulse/logcollector/PulseKafkaConsumer.scala]: Does it make sense to have an application name as `""`? Or should this be an error?

Comment (afoerster) [reply log-collector/src/main/scala/io/phdata/pulse/logcollector/PulseKafkaConsumer.scala]: No it doesn't.  I thought about what to do with this, it should either be just thrown away or logged. We want to figure out how to debounce logging before we do that though. Also only way this code path can be reached is through the Kafka appender, so it's not an immediate concern. I'll add a task for a future commit to fix this.

**Comment by afoerster** [comment]
I'm having some issues getting the tests to pass, I think it has to do with the way Travis does thread scheduling. They pass locally. I've tried increasing sleep times with no success.

**Comment by afoerster** [comment]
I gave up on trying to get all the tests to run on the build server. I tried removing some parallelism in the stream and increasing timeouts. I think running solr, zookeeper, kafka, and this application all on the two cores Travis provides might just be too much. Either way I've spent enough time on it.

The tests are now in an integration test project, which is where they probably belong anyways. It would be good to get some unit tests around the same functionality using mocks.

File: log-collector/src/main/scala/io/phdata/pulse/logcollector/PulseKafkaConsumer.scala (hunk 2 above)
Comment (afoerster) [reply log-collector/src/main/scala/io/phdata/pulse/logcollector/PulseKafkaConsumer.scala]: @safwanislam can you add a TODO comment in your next commit to address this issue?

**Pull request merged**